src/plist.h

```diff
@@ -238,8 +238,29 @@ extern void            AssPlist (
 
 /****************************************************************************
 **
+*F  ADD_PLIST( <list>, <val> ) . . . . . . . .  assign to end of a plain list
+**
+**  Note that this function does not adjust the TNUM of the list object. It
+**  also does not attempt to convert the list to a different representation,
+**  such as a string or blist. If your need that, use AddList or AddPlist
+**  instead.
+**
+*/
+static inline UInt ADD_PLIST(Obj list, Obj val)
+{
+    const UInt pos = LEN_PLIST(list) + 1;
+    GROW_PLIST( list, pos );
+    SET_LEN_PLIST( list, pos );
+    SET_ELM_PLIST( list, pos, val );
+    if (IS_BAG_REF(val))
```

Comment:
Anyway, I'll remove this optimization -- but I'll do so in PR #1605, where this change logically belongs. Once that PR has been merged, I can then rebase this PR.